app/src/main/java/edu/uc/forbesne/shoppingsidekick/ui/main/MainFragment.kt

```diff
@@ -37,6 +38,7 @@ class MainFragment : Fragment() {
 
         viewModel.fetchAllProducts()
         viewModel.products.observe(this, Observer { products ->
+            val actProductName = null
```

Comment:
I updated within the viewModel to add a null along with a val from the suggestion of:https://kotlinlang.org/docs/coding-conventions.html#control-flow-statements

Reply:
The concept of val and null clash.  
- A val cannot be reassigned.
- It's currently assigned to null.
So, that means this will always be null, and thus, will only cause a NullPointerException.  It cannot hold an object.

Is it used anywhere?

This appears to increase, not decrease, technical debt.